trie/database.go

```diff
@@ -662,10 +662,12 @@ func (db *Database) Commit(node common.Hash, report bool, callback func(common.H
 	}
 	// Uncache any leftovers in the last batch
 	db.lock.Lock()
-	defer db.lock.Unlock()
-
-	batch.Replay(uncacher)
+	err := batch.Replay(uncacher)
 	batch.Reset()
+	defer db.lock.Unlock()
```

Comment:
Please move this up again -- it's nice to keep the `Lock()` and `defer Unlock()` right after eachother, it makes it obvious that the pair is complete. And in this instance it doesn't seem to be any particular reason to not do that. 

Plus it makes the diff smaller. IMO the section here could look like this:

```golang
db.lock.Lock()
defer db.lock.Unlock()
if err := batch.Replay(uncacher); err != nil{
    return err
}
batch.Reset
```